src/main/groovy/uk/ac/sanger/scgcf/jira/services/actions/TransferActions.groovy

```diff
@@ -83,86 +72,121 @@ class TransferActions {
 
         validateCombineParameters(sourceLabwares, destinationLabware)
 
-        (0..sourceLabwares.size() - 1).each { plateNumber ->
-            def sourceLabware = sourceLabwares[plateNumber]
-            def transferMap = createTransferMapForCombine(sourceLabware.labwareType.layout, plateNumber)
-
-            destinationLabware = transfer(sourceLabware, destinationLabware,
-                newMetadataToLocation, materialType, copyMetadata, transferMap)
+        def transferMap = []
+        sourceLabwares.eachWithIndex { sourceLabware, plateNumber ->
+            def layout = sourceLabware.labwareType.layout
+
+            transferMap += (1..layout.row).collect { row ->
+                (1..layout.column).collect { column ->
+                    def sourceRow = (char) (64 + row)
+                    def sourceColumn = column
+
+                    def destinationRow = (char) (64 + (row * 2) - (((int) (plateNumber / 2)) ? 0 : 1))
+                    def destinationColumn = column * 2 - (plateNumber % 2 ? 0 : 1)
+
+                    new TransferMapping(
+                        sourceBarcode: sourceLabware.barcode,
+                        sourceLocation: "$sourceRow$sourceColumn",
+                        destinationLocation: "$destinationRow$destinationColumn",
+                        destinationBarcode: destinationLabware.barcode
+                    )
+                }
+            }.flatten()
         }
-
-        updateLabware(destinationLabware)
+        def destinationLabwares = transfer(sourceLabwares, [destinationLabware], newMetadataToLocation, materialType, copyMetadata, transferMap)
+        destinationLabwares[0]
     }
 
     def static pool(Labware sourceLabware, Labware destinationLabware,
                     MaterialType materialType, List<String> locations,
                     List<Metadatum> newMetadata = []) {
 
-        if (destinationLabware.labwareType.name != 'generic tube') {
-            throw new TransferException('The destination labware should be a tube')
+        if (destinationLabware.labwareType.name != LabwareTypes.GENERIC_TUBE.name) {
```

Comment:
I haven't realised this LabwareTypes enum got into the code. I still have a bad feeling about it. If somebody change these values in the DB, then it won't work. It is not DRY.